### ISSUE TITLE
"None should be instance of 'str'" when following Tutorial

### ISSUE THREAD

**Issue by dginovker** [AppImageCrafters/appimage-builder#63]
Following this tutorial here: https://appimage-builder.readthedocs.io/en/latest/intro/tutorial.html

After setting up the repository and running the app during `appimage-builder --generate` just fine, my next step was (to fix docker) and then afterwards run this command: `appimage-builder --skip-test --skip-appimage`

Which produced:

```
dr-kracket@dr-kracket:~/Downloads/qt-appimage-template$ appimage-builder --skip-test --skip-appimage
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/schema.py", line 393, in validate
    nvalue = Schema(svalue, error=e, ignore_extra_keys=i).validate(value)
  File "/usr/local/lib/python3.8/dist-packages/schema.py", line 428, in validate
    raise SchemaUnexpectedTypeError(message, e.format(data) if e else None)
schema.SchemaUnexpectedTypeError: None should be instance of 'str'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/schema.py", line 431, in validate
    return s.validate(data)
  File "/usr/local/lib/python3.8/dist-packages/schema.py", line 397, in validate
    raise SchemaError([message] + x.autos, [e] + x.errors)
schema.SchemaError: Key 'arch' error:
None should be instance of 'str'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/schema.py", line 393, in validate
    nvalue = Schema(svalue, error=e, ignore_extra_keys=i).validate(value)
  File "/usr/local/lib/python3.8/dist-packages/schema.py", line 433, in validate
    raise SchemaError([None] + x.autos, [e] + x.errors)
schema.SchemaError: Key 'arch' error:
None should be instance of 'str'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/appimage-builder", line 8, in <module>
    sys.exit(__main__())
  File "/usr/local/lib/python3.8/dist-packages/appimagebuilder/__main__.py", line 81, in __main__
    recipe_data = load_recipe(args.recipe)
  File "/usr/local/lib/python3.8/dist-packages/appimagebuilder/__main__.py", line 146, in load_recipe
    recipe_validator.v1.validate(recipe_data)
  File "/usr/local/lib/python3.8/dist-packages/schema.py", line 397, in validate
    raise SchemaError([message] + x.autos, [e] + x.errors)
schema.SchemaError: Key 'AppImage' error:
Key 'arch' error:
None should be instance of 'str'
```

Running since the commit that fixed `allow_unauthenticated`

*****

OS: Ubuntu 20.04.1 LTS x86_64 
Kernel: 5.4.0-48-generic

**Comment by azubieta** [comment]
This, provably because there is an empty entry in the apt 'include' or 'exclude' list (or somewhere else). It should look like this:
```
apt:
  include:
  - 
```

**Comment by dginovker** [comment]
Sorry, I should have included my yml :smile:. There's no empty fields:

```
# appimage-builder recipe see https://appimage-builder.readthedocs.io for details
version: 1
AppDir:
  path: ./AppDir
  app_info:
    id: QtQuickControls2Application
    name: SimpleQtApp
    icon: QtQuickControls2Application
    version: latest
    exec: usr/bin/qt-appimage-template
    exec_args: $@
  runtime:
    env:
      APPDIR_LIBRARY_PATH: $APPDIR/lib/x86_64-linux-gnu
  apt:
    arch: amd64
    allow_unauthenticated: true
    sources:
    - sourceline: deb http://us.archive.ubuntu.com/ubuntu/ focal main restricted
    - sourceline: deb http://us.archive.ubuntu.com/ubuntu/ focal-updates main restricted
    - sourceline: deb http://us.archive.ubuntu.com/ubuntu/ focal universe
    - sourceline: deb http://us.archive.ubuntu.com/ubuntu/ focal-updates universe
    - sourceline: deb http://us.archive.ubuntu.com/ubuntu/ focal multiverse
    - sourceline: deb http://us.archive.ubuntu.com/ubuntu/ focal-updates multiverse
    - sourceline: deb http://us.archive.ubuntu.com/ubuntu/ focal-backports main restricted
        universe multiverse
    - sourceline: deb http://security.ubuntu.com/ubuntu focal-security main restricted
    - sourceline: deb http://security.ubuntu.com/ubuntu focal-security universe
    - sourceline: deb http://security.ubuntu.com/ubuntu focal-security multiverse
    - sourceline: deb [arch=amd64] https://download.docker.com/linux/ubuntu focal
        stable
    - sourceline: deb [arch=amd64] https://packages.microsoft.com/ubuntu/20.04/prod
        focal main
    - sourceline: deb https://download.mono-project.com/repo/ubuntu stable-focal main
    include:
    - libdbus-1-3
    - libexpat1
    - libgpg-error0
    - liblzma5
    - libpcre3
    - libqt5gui5
    - libsystemd0
    - libuuid1
    exclude: []
  files:
    exclude:
    - usr/share/man
    - usr/share/doc/*/README.*
    - usr/share/doc/*/changelog.*
    - usr/share/doc/*/NEWS.*
    - usr/share/doc/*/TODO.*
  test:
    fedora:
      image: appimagecrafters/tests-env:fedora-30
      command: ./AppRun
      use_host_x: true
    debian:
      image: appimagecrafters/tests-env:debian-stable
      command: ./AppRun
      use_host_x: true
    arch:
      image: appimagecrafters/tests-env:archlinux-latest
      command: ./AppRun
      use_host_x: true
    centos:
      image: appimagecrafters/tests-env:centos-7
      command: ./AppRun
      use_host_x: true
    ubuntu:
      image: appimagecrafters/tests-env:ubuntu-xenial
      command: ./AppRun
      use_host_x: true
AppImage:
  arch: null
  update-information: None
  sign-key: None
```

(At least http://www.yamllint.com/ says it's a valid yaml :eyes:)

**Comment by azubieta** [comment]
It seems that we didn't read properly the error message, it complains about the arch key which is set to `null`. It should be `arch: x86_64`  instead.

**Comment by azubieta** [comment]
This should be fixed in with latest release (v0.8.2).
Feel free to reopen the issue if required.
